fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/common/Utils.java

```diff
@@ -257,5 +257,19 @@ public static String performServerOutputTemplateLocationGet(final RequestSpecifi
                 .andReturn().asString();
         return templateLocation.substring(1,templateLocation.length()-1);
     }
+
+    public static void waitUntilFileCreation(String location){
+        Long start= System.currentTimeMillis();
+
+        while (!new File(location).exists()) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
```

Comment:
don't do this, instead simply add `throws InterruptedException`  to progagate this up through until the test.